### ISSUE TITLE
vsphere clone bootstrapping via winrm not able to use ipv6 address with port

### ISSUE THREAD

**Issue by CEI-Rwitt** [chef-partners/knife-vsphere#231]
Hello,
Apologies if I'm jumping the gun by creating this issue. I have searched the issues (open & closed), chef discourse, and google but have yet to find something that relates directly to my issue.

I'm in an almost entirely windows environment. In this example I'm using a win 10 client to clone and bootstrap a windows vm on vsphere/vcenter/esxi 5.1. 
`knife vsphere vm clone ACloneName --folder 'ATemplateSourceFolder' --template 'A windows template name' --dest-folder 'ADestinationFolder' --cspec 'A Customization Spec for win10vm in domain' --bootstrap --run-list 'role[windows_base]'`

Specifically, my issue is that during the bootstrapping portion (after the clone is customized) some of the time the IP address returned will be the IPv6 address. When this happens the bootstrap fails to connect  due to InvalidURIError. I believe this is because it is trying to connect to IPv6Address:Port rather than containing the address separately from the port like [IPv6Address]:Port. For instance I believe this is incorrect:
`fe80::84c9:8e21:6cc9:8cad:5985`
I think this should work but is not the way the URI is being formatted:
`[fe80::84c9:8e21:6cc9:8cad]:5985`

I have verified that I can ping the IPV6 address and connect to `http://[fe80::84c9:8e21:6cc9:8cad]:5985/wsman' remotely.

Here is a sample of the failing procedure:

> Cloning template A windows template name to new VM ACloneName
> Finished creating virtual machine ACloneName
> Powered on virtual machine ACloneName
> Waiting for customization to complete...
> Waiting for sysprep...............
> Customization of VM ACloneName succeeded. Customization log located at C:\Windows\TEMP\vmware-imc\guestcust.log in the guest OS.
> Customization Complete
> 
> Waiting for winrm access to become available on fe80::84c9:8e21:6cc9:8cad:5985......done
> WARNING: Unsupported version of httpclient. The supported major version of library is httpclient version 2.6. This code path monkey patches few methods in httpclient to support additional features. > This could possibly work, but it is advised to extensively test your version. If you are aware of the impact of using 2.7.0, this warning can be disabled by setting USE_HTTPCLIENT_MAJOR to the major version 2.7. 
> ERROR: URI::InvalidURIError: the scheme http does not accept registry part: fe80::84c9:8e21:6cc9:8cad:5985 (or bad hostname?)

Most of the time the IPv4 address will return and not cause an issue.

I wonder:
- Is it currently possible to force the clone + bootstrap to always use the ipv4 address? I have not found an option that seems to address this.
- Does anyone have another work-around suggestion? For this case I can switch to using a pre-determined static ip address but I'd rather not.

**Comment by swalberg** [comment]
Hi, thanks for the detailed bug report! 

We're getting that address from the VMWare tools and if memory serves we only get one (that is, there's nothing to choose from). It's a v6 link local address, so you're on the same network segment? Does the box pick up a DHCP address? What about the address reported in the summary page?

My guess is that this is a race condition where we poll VMWare for the address and we catch it when IPv6 is up but DHCP hasn't finished.

Do you have a FQDN of the box by chance? `--fqdn` may help. Otherwise I'd have to trace around a bit. I think this may be inside of Chef as the "Waiting for..." shows that we're able to talk to it.

**Comment by CEI-Rwitt** [comment]
Yes, the client and destination are on the same segment and winrm connection works fine outside of knife and knife-vsphere. IPv4 is via DHCP.

I'll try the fqdn. Thanks for your assistance!

**Comment by swalberg** [comment]
Hi, did you ever get this to work? Since you've filed this issue, we've added `--bootstrap-ipv4` which will force it to wait until it gets an actual v4 address instead of the link local v6 one.
